content/courses/Ada_For_The_C_Embedded_Developer/chapters/04_Embedded.rst

```diff
@@ -782,7 +782,85 @@ Previously, we've seen that we can use
 layout for a record type. As mentioned before, this is useful when interfacing
 with hardware, drivers, or communication protocols. In this section, we'll
 extend this concept for two specific use-cases: register overlays and data
-streams.
+streams. Before we discuss those use-cases, though, we'll first explain the
+:ada:`Size` aspect and the :ada:`Size` attribute.
+
+.. _Size_Aspect_Attribute:
+
+:ada:`Size` aspect and attribute
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :ada:`Size` aspect indicates the minimum number of bits required to
+represent an object, and it can be used to confirm expectations. When applied
+to a type, the :ada:`Size` aspect is telling the compiler to not make record or
+array components of a type :ada:`T` any smaller than :ada:`X` bits. Therefore,
+a common usage for this aspect is to just confirm expectations: developers
+specify :ada:`'Size` to tell the compiler that :ada:`T` should fit :ada:`X`
+bits, and the compiler will tell them if they are right (or wrong).
+
+When the specified size value is larger than necessary, it can cause objects to
+be bigger in memory than they would be otherwise. For example, for some
+enumeration types, we could say :ada:`for type Enum'Size use 32;` when the
+number of literals would otherwise have required only a byte. That's useful for
+unchecked conversions because the sizes of the two types need to be the same.
+Likewise, it's useful for interfacing with C, where :c:`enum` types are just
+mapped to the :ada:`int` type, and thus larger than Ada might otherwise
+require. We'll discuss unchecked conversions
+:ref:`later in the course <OverlaysVsUncheckedConversions>`.
+
+Let's look at an example from an earlier chapter:
+
+:code-config:`accumulate_code=True`
+
+[Ada]
+
+.. code:: ada compile_button project=Courses.Ada_For_C_Embedded_Dev.Perspective.Size_Aspect
+
+    package My_Device_Types is
+
+       type UInt10 is mod 2 ** 10
+         with Size => 10;
+
+    end My_Device_Types;
+
+Here, we're saying that objects of type :ada:`UInt10` must have at least 10
```

Comment:
I'm reusing your comments (from https://github.com/AdaCore/learn/pull/484#discussion_r554072004) here and below.